tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts

```diff
@@ -270,7 +273,7 @@ export class AnnotationComponent implements AfterViewInit, OnChanges {
         }.bind(this)
       )
       .attr('class', `hint-text ${this.textClass}`)
-      .text((d: string) => d);
+      .text((d: string) => d.split('/').slice(1).join());
```

Comment:
Instead of this, you may want to get the displayName for a run separate from the id. In other words, don't derive the display name from the id.

Reply:
I think I dont completely understand. This is what happens in the metrics plugin:
```
function parseRunId(runId: string): {run: string; experimentId: string} {
  const slashIndex = runId.indexOf('/');
  return {
    run: runId.substring(slashIndex + 1),
    experimentId: runId.substring(0, slashIndex),
  };
}
```
How should I get the runId instead?

Reply:
Metrics plugin is doing it right: it is defining that function and is being only used in the data source. Data source is the layer where we are more aware of a specific backend logic and in that case, we needed to derive runId and runDisplayName. 

I will put it in another way. Previously, there was no difference between runId vs. run display name; everything was just a string and they were not unique (you've already handled this in the data source). Now, in the experiment aware Angular code, we now differentiate name vs. id explicitly and regard the runId as something opaque. Data structure like the ValueData should only use the `runId` and never hold onto the `runName`. Here, the issue is, `runId`, depending on a backend, can be literally just numbers or random bytes, not composed of `eid/rid` yet here, you are extracting the name from this opaque id. 

Now, moving on to remediation: runSelector provides a data structure that gives you more metadata about a run including its displayName. You should use it to grab the data from the container, form a data structure, and pass it down to this component. I suspect it will be a large for this already large PR so I recommend fixing it in a quick follow up.

What do you think?

Reply:
Sure, sounds good. Now I think I also finally fully understand the differentiation between runName and runId. 

To clarify for this next PR, after looking into `tensorboard/webapp/runs/store/runs_selectors.ts`, there is no "get metadata for all runs", so I assume I would have to fetch all active runs and then use this:
```
/**
 * Returns Observable that emits run object.
 */
export const getRun = createSelector(
  getDataState,
  (state: RunsDataState, props: {runId: string}): Run | null => {
    return state.runMetadata[props.runId] ?? null;
  }
);
```
to get the run information for all the IDs?